w07-ruby-on-rails-cont/d3-homework/README.md

```diff
@@ -0,0 +1,88 @@
+
+Use the following collaboration scenarios to do the following with a partner.
+
+1. Create a rails project, push it to github, and make your partner a collaborator. (2 projects total!)
+1. On your own project add an index page for a single resource - whatever you like. Get creative.
+1. Clone the project your partner created, make a new branch, add a splash page to it and make the branch into a pull request.
+1. On your own project in github, accept and merge your partner's pull request.
+
+You should both have a project with a splash page and a `posts#index` page.
+
+## Collaborator Scenario
+
+Imagine you and Jessica, Sally, and John want to build an app together. John bootstraps a rails project and pushes it to Github and adds the other three as <a href="https://help.github.com/articles/adding-collaborators-to-a-personal-repository/" target="_blank">collaborators</a>. You divvy up tasks and set to work. Your first task is to add user authentication.  At the same time, Jessica is doing API integration, Sally is writing tests, and John is going to be git master and put together a style guide and the splash page.
+
+1. You clone the project John pushed up to Github after he makes you a collaborator. This creates a local repository with a `remote` called `origin` which is the original repo John created and that is technically owned by John. From within your local repository, you can check the urls of all remote repos it's connected to.
+  ```
+  $ git remote -v
+  ```
+
+2. Make a new branch called `auth`. Feature branches help organize work on a project.
+  ```
+    $ git checkout -b auth
+  ```
+
+3. Now you build the entire auth feature with sign up, login, logout, etc. How do you get your new code up to the main group project?
+
+4. Switch back to the local/master branch, then pull from origin/master to bring your local master up to date.
+  ```
+    $ git checkout master
+    $ git pull origin master
+  ```
+
+5. Now switch back to your local/auth branch and merge local/master into it. This will give you a chance to resolve all conflicts with the most up to date master version before pushing your branch to origin/master. Watch git's output in your terminal to see which files have merge conflicts.
+  ```
+    $ git checkout auth
+    $ git merge master
+  ```
+
+6. Once you have resolved all the conflicts, add and commit your changes locally and push your feature branch to the remote repo.
```

Comment:
I assume you're going over this in detail in the lecture.  Cause it's sooo not easy.